modules/apps/asset/asset-publisher-web/src/main/java/com/liferay/asset/publisher/web/internal/display/context/AssetPublisherDisplayContext.java

```diff
@@ -2130,8 +2136,10 @@ private String _getSegmentsAnonymousUserId() {
 	}
 
 	private long[] _getSegmentsEntryIds() {
-		return GetterUtil.getLongValues(
-			_portletRequest.getAttribute(SegmentsWebKeys.SEGMENTS_ENTRY_IDS));
+		return _segmentsEntryRetriever.getSegmentsEntryIds(
+			_themeDisplay.getScopeGroupId(), _themeDisplay.getUserId(),
+			_requestContextMapper.map(
+				PortalUtil.getHttpServletRequest(_portletRequest)));
```

Comment:
I see a lot of uses of `PortalUtil` in the class, I guess It's ok to use one more time